principles/solid/dependencyInjection.md

```diff
@@ -1,22 +1,58 @@
 ## Dependency Injection
 
-[TODO]
-
+The dependency injection principle suggests that objects should receive dependencies from external sources instead of creating them internally. 
 ### Bad example
 
-[TODO]
+Suppose you have a UserService class that requires a UserRepository object to perform database operations. Instead of injecting the dependency, the UserService class creates the UserRepository object internally:
+``` java
+public class UserService {
+	private UserRepository userRepository = new UserRepository();
+
+	public void addUser(User user) {
+		userRepository.save(user);
+	}
+
+	public User getUser(String email) {
+		return userRepository.findByEmail(email);
+	}
+}
 
+```
+In this example, the UserService class is tightly coupled to the UserRepository class. If the implementation of the UserRepository changes, the UserService class would also need to change.
 ### Good example
 
-[TODO]
+To apply the Dependency Injection principle, you would inject the UserRepository dependency into the UserService class:
+``` java
+public class UserService {
+	private UserRepository userRepository;
```

Comment:
Change to an interface, and explain why it's a  better option using interfaces

Reply:
The description of all the principles must include the problem, how it solves, and the benefits of applying it.